internal/service/indexer/indexer.go

```diff
@@ -27,54 +30,66 @@ const (
 	fabricDeployEventSignature   = "TokenContractDeployed(address,(uint256,string,string))"
 )
 
-var logsHandlers = map[string]func(i *Indexer, eventLog types.Log, client *ethclient.Client) error{
-	crypto.Keccak256Hash([]byte(issuerTransferEventSignature)).Hex(): (*Indexer).handleIssuerTransferLog,
-	crypto.Keccak256Hash([]byte(fabricDeployEventSignature)).Hex():   (*Indexer).handleFabricDeployLog,
+var logsHandlers = map[string]func(i *indexer, eventLog types.Log, client *ethclient.Client) error{
+	crypto.Keccak256Hash([]byte(issuerTransferEventSignature)).Hex(): (*indexer).handleIssuerTransferLog,
+	crypto.Keccak256Hash([]byte(fabricDeployEventSignature)).Hex():   (*indexer).handleFabricDeployLog,
 }
 
-func NewIndexer(cfg config.Config, ctx context.Context, addresses []string, blocks []int64, cancel context.CancelFunc) IIndexer {
-	return &Indexer{
-		cfg:        cfg,
-		ctx:        ctx,
-		log:        cfg.Log(),
-		Addresses:  addresses,
-		Blocks:     blocks,
-		ContractsQ: postgres.NewContractsQ(cfg.DB().Clone()),
-		Cancel:     cancel,
+func NewIndexer(cfg config.Config, ctx context.Context, addresses []string, blocks []int64, cancel context.CancelFunc, wg *sync.WaitGroup) Indexer {
+	return &indexer{
+		cfg:             cfg,
+		ctx:             ctx,
+		log:             cfg.Log(),
+		Addresses:       addresses,
+		Blocks:          blocks,
+		ContractsQ:      postgres.NewContractsQ(cfg.DB().Clone()),
+		UsersQ:          postgres.NewUsersQ(cfg.DB().Clone()),
+		Cancel:          cancel,
+		Clients:         map[string]*ethclient.Client{},
+		CertIntegrators: map[string]*contracts.CertIntegratorContract{},
+		dailyStorage:    storage.DailyStorageInstance(ctx),
+		wg:              wg,
 	}
 }
 
-func (i *Indexer) Run(ctx context.Context) {
+func (i *indexer) Run(ctx context.Context) {
 	go running.WithBackOff(
```

Comment:
Why `running.WithBackOff`? For what do you need that 1 second for restart? Use `for` cycle